flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/types/LocationPreferenceSlotSelectionStrategyTest.java

```diff
@@ -63,67 +54,81 @@ public void testPhysicalSlotResourceProfileRespected() {
-                        .isMatching(slotProfile.getPhysicalSlotResourceProfile()));
+        assertThat(match)
+                .hasValueSatisfying(
+                        slotInfoAndLocality ->
+                                assertThat(
+                                                slotInfoAndLocality
+                                                        .getSlotInfo()
+                                                        .getResourceProfile()
+                                                        .isMatching(
+                                                                slotProfile
+                                                                        .getPhysicalSlotResourceProfile()))
+                                        .isTrue());
 
         ResourceProfile evenBiggerResourceProfile =
                 ResourceProfile.fromResources(
                         biggerResourceProfile.getCpuCores().getValue().doubleValue() + 1.0,
                         resourceProfile.getTaskHeapMemory().getMebiBytes());
-        slotProfile =
+        final SlotProfile slotProfileNotMatching =
                 SlotProfile.priorAllocation(
                         resourceProfile,
                         evenBiggerResourceProfile,
                         Collections.emptyList(),
                         Collections.emptyList(),
                         Collections.emptySet());
 
-        match = runMatching(slotProfile);
-        Assert.assertFalse(match.isPresent());
+        match = runMatching(slotProfileNotMatching);
+        assertThat(match).isNotPresent();
     }
 
     @Test
-    public void matchNoRequirements() {
+    void matchNoRequirements() {
 
         SlotProfile slotProfile = SlotProfileTestingUtils.noRequirements();
         Optional<SlotSelectionStrategy.SlotInfoAndLocality> match = runMatching(slotProfile);
 
-        Assert.assertTrue(match.isPresent());
-        final SlotSelectionStrategy.SlotInfoAndLocality slotInfoAndLocality = match.get();
-        assertThat(candidates, hasItem(withSlotInfo(slotInfoAndLocality.getSlotInfo())));
-        assertThat(slotInfoAndLocality, hasLocality(Locality.UNCONSTRAINED));
+        assertThat(match)
+                .hasValueSatisfying(
+                        slotInfoAndLocality -> {
+                            assertThat(slotInfoAndLocality.getLocality())
+                                    .isEqualTo(Locality.UNCONSTRAINED);
+                            assertThat(candidates)
+                                    .anySatisfy(
+                                            slotInfoAndResources ->
+                                                    assertThat(slotInfoAndResources.getSlotInfo())
+                                                            .isEqualTo(
+                                                                    slotInfoAndLocality
+                                                                            .getSlotInfo()));
+                        });
```

Comment:
I'd prefer extract this logic to a specific method, this will also be helpful for adding new test case in the future.